src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,174 @@
+using LinearAlgebra: svd, eig, eigen
+
+@doc doc"""
+    SymmetricPositiveDefinite{N} <: Manifold
+
+The manifold of symmetric positive definite matrices, i.e.
+
+```math
+\mathcal P(n) \coloneqq
+\bigl\{
+    x \in \mathbb R^{n\\times n} :
+    \xi^\mathrm{T}x\xi > 0 \text{ for all } \xi \in \mathbb R^{n}\backslash\{0\}
+\bigr\}
+```
+
+# Constructor
+
+    SymmetricPositiveDefinite(n)
+
+generates the $\mathcal P(n) \subset \mathbb R^{n\times n}$
+"""
+struct SymmetricPositiveDefinite{N} <: Manifold end
+SymmetricPositiveDefinite(n::Int) = SymmetricPositiveDefinite{n}()
+
+@doc doc"""
+    manifold_dimension(::SymmetricPositiveDefinite{N})
+
+returns the dimension of the manifold [`SymmetricPositiveDefinite`](@ref) $\mathcal P(n)$, N\in \mathbb N$, i.e.
+```math
+    \frac{n(n+1)}{2}    
+```
+"""
+@generated manifold_dimension(::SymmetricPositiveDefinite{N}) where {N} = N*(N+1)/2
+
+@doc doc"""
+
+"""
+struct LinearAffineMetric <: Metric end
+
+@doc doc"""
+
+"""
+struct LogEuclideanMetric <: Metric end
+
+distance(P::SymmetricPositiveDefinite{N},x,y) = distance(MetricManifold(P,LinearAffineMetric),x,y)
+function distance(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,y)
```

Comment:
Don't you need a `where {N}` when you mention the parameter explicitly? Similarly below.

Reply:
Yes, I fixed that.